src/Cli/dotnet/Commands/Project/Convert/ProjectConvertCommand.cs

```diff
@@ -47,11 +51,71 @@ public override int Execute()
             File.Move(file, targetFile);
         }
 
+        // Create project file.
         string projectFile = Path.Join(targetDirectory, Path.GetFileNameWithoutExtension(file) + ".csproj");
         using var stream = File.Open(projectFile, FileMode.Create, FileAccess.Write);
         using var writer = new StreamWriter(stream, Encoding.UTF8);
         VirtualProjectBuildingCommand.WriteProjectFile(writer, directives, isVirtualProject: false);
 
+        // Copy over included items.
+        foreach (var item in includeItems)
+        {
+            string targetItemFullPath = Path.Combine(targetDirectory, item.RelativePath);
+
+            // Ignore already-copied files.
+            if (File.Exists(targetItemFullPath))
+            {
+                continue;
+            }
+
+            string targetItemDirectory = Path.GetDirectoryName(targetItemFullPath)!;
+            Directory.CreateDirectory(targetItemDirectory);
+            File.Copy(item.FullPath, targetItemFullPath);
+        }
+
         return 0;
+
+        IEnumerable<(string FullPath, string RelativePath)> FindIncludedItems()
+        {
+            string entryPointFileDirectory = PathUtility.EnsureTrailingSlash(Path.GetDirectoryName(file)!);
+            var projectCollection = new ProjectCollection();
+            var command = new VirtualProjectBuildingCommand(
+                entryPointFileFullPath: file,
+                msbuildArgs: [])
+            {
+                Directives = directives,
+            };
+            var projectInstance = command.CreateProjectInstance(projectCollection);
+
+            // Include only items we know are files.
+            string[] itemTypes = ["Content", "None", "Compile", "EmbeddedResource"];
+            var items = itemTypes.SelectMany(t => projectInstance.GetItems(t));
+
+            foreach (var item in items)
+            {
+                // Escape hatch - exclude items that have metadata `ExcludeFromFileBasedAppConversion` set to `true`.
+                string include = item.GetMetadataValue("ExcludeFromFileBasedAppConversion");
```

Comment:
Is this an escape hatch for us or for users? If it's for users, how do they set this metadata if there's no project file?

Reply:
Both, but I can remove it, don't have any strong motivation for it. Could be set via Directory.Build.props or nuget packages can set it for their items, etc.